foundry/applications/formApplications/imagePopout.d.ts

```diff
@@ -1,10 +1,96 @@
 /**
- * An Image Popout Application
- * Provides optional support to edit the image path being viewed
- * @param image - The image being viewed
- * @param options - Standard Application rendering options
- * @param onUpdate - An optional callback function which should be triggered if the Image path is edited
+ * An Image Popout Application which features a single image in a lightbox style frame.
+ * This popout can also be used as a form, allowing the user to edit an image which is being used.
+ * Furthermore, this application allows for sharing the display of an image with other connected players.
+ *
+ * @example
+ * ```typescript
+ * // Construct the Application instance
+ * const ip = new ImagePopout("path/to/image.jpg", {
+ *   title: "My Featured Image",
+ *   shareable: true,
+ *   entity: game.actors.getName("My Hero")
+ * });
+ *
+ * // Display the image popout
+ * ip.render(true);
+ *
+ * // Share the image with other connected players
+ * ip.share();
+ * ```
  */
-declare class ImagePopout extends FormApplication {
-  _updateObject(event?: Event, formData?: object): Promise<any>; // TODO: Add correct return type
+declare class ImagePopout extends FormApplication<ImagePopout.Data, string> {
+  constructor(src: string, options?: ImagePopout.Options);
+
+  protected _related: Entity | object | null;
+
+  /** @override */
+  static get defaultOptions(): ImagePopout.Options;
+
+  /** @override */
+  get title(): string;
+
+  /** @override */
+  getData(options?: Application.RenderOptions): ImagePopout.Data | Promise<ImagePopout.Data>;
+
+  /**
+   * Test whether the title of the image popout should be visible to the user
+   */
+  isTitleVisible(): boolean;
+
+  /**
+   * Provide a reference to the Entity referenced by this popout, if one exists
+   */
+  getRelatedObject(): Promise<Entity | object | null>;
+
+  /** @override */
+  protected _render(force?: boolean, options?: Application.RenderOptions): Promise<void>;
```

Comment:
This method sets `this.position` with the return type of `getPosition`, which might not be a full `Application.Position`. That should be explicitly typed and if need be, changed in `Application`.

Reply:
Well, I found out that I need to to the same on my previous class (journalSheet), as it uses that `getPosition` inside and also sets its `this.position` to that.

I'm not sure how many times the `Application.Position` will be used throughout the application - maybe `scale` property can be marked as optional. How about sticking with:
```typescript
type ImagePopuout.Position = {width: number; height: number; top: number; left: number;} | {width: number; height: number}
```
with `todo` to check at the end of 0.7.9 migration if we can mark `scale` as optional and put `Application.Position` as first parameter in union above. What do you think, @kmoschcau?

Reply:
At that point I would just make `top`, `left` and `scale` of `Application.Position` optional and use that everywhere. If it can then be said for a return value that the properties are definitely said in one case, you can use `Required<Application.Position>` for example. @ghost91- @BoltsJ what do you guys think?

Reply:
I think `scale` can always be optional, so using `Required` on the return type of `setPosition` would not work:
```js
game.actors.getName('Gorgoth').sheet.setPosition({height: 'auto'})
{width: 745, height: 1106, left: 185.5, top: 23, scale: undefined}
```

Reply:
In that case, maybe I just make the `Application.Position.scale` optional and stick with
```typescript
type ImagePopout.Position = Application.Position | {width: number, height: number}
```
The question is how many other classes have that `incomplete position` inside. I can see that it is the case in both `ImagePopout` and `JournalSheet`. Maybe that type above is not a bad idea, as it can be just an exception to the rule. @kmoschcau @ghost91- @BoltsJ ?

Reply:
@Takitaf I'd need to take a closer look at this first to completely understand what is going on. Does it make sense for me to dive deeply into this right now or will you be able to handle this without me? :D What is the state of this PR in general? Just want to make sure my (already quite limited) time is well invested :)

Reply:
@ghost91- Well, I think everything is done on my local environment, except this one. 

As far as I can see (I don't know whole codebase well, at least for now, unfortunately), this class has narrower `position` field type (only `width` and `height` are set). Either we can change the `Application.Position` type by making `scale`, `top` and `left` properties optional, or we can make union type on `position` field `Application.Position | {width: number, height: number}`. Both changes can potentially make some problems in a future probably.

Anyway, I wanted to commit cr fixes, but someone updated `husky` lib and it seems to not working properly (at least on my local env with Windows), so I need to find a solution to that in order to be able to commit changes.

Reply:
OK, I will take a look then. We should be able to find a solution. It would be great if we coud move forward with this.

Regarding your issue with husky, feel free to ping me in the League Discord (I'm called "ghost" there), maybe we can figure it out together.

Reply:
You could do it like this `Pick<Application.Position, 'width' | 'height'>`, or even this `Partial<Application.Position> & Pick<Application.Position, 'width' | 'height'>`.